gpiozero/pins/mock.py

```diff
@@ -413,3 +385,34 @@ def tx_word(self, value, bits_per_word=None):
             bits = reversed(bits)
         self.tx_buf.extend(bits)
 
+
+class MockFactory(LocalPiFactory):
+    def __init__(self, revision='a21041', pin_class=MockPin):
+        super(MockFactory, self).__init__()
+        self._revision = revision
+        self.pin_class = pin_class
+
+    def _get_address(self):
+        return ('mock',)
+
+    def _get_revision(self):
+        return self._revision
+
+    def reset(self):
```

Comment:
The bit that concerned me was seeing `Device._pin_factory.reset()` which kinda implies that all Device pin_factories have a reset() method?

I think it's something I might have touched on before, but it would be great if some/all of the MockPin unit tests could also (somehow) be (manually?) run using real pins on a RPi.